scripts/make-dev-releases

```diff
@@ -34,25 +34,120 @@
 
 set -e
 
+# Log a line to sterr and exit with error code 1
+err() {
+  echo "ERROR: $1" >&2
+  exit 1
+}
+
 if [ $# -eq 0 ]; then
   read -r -p "Please enter the wanted version number (example: 4.12.1): " version
+  echo ""
   if [ -z "${version}" ]; then
-    echo "Please enter a valid version number"
-    exit 1
+    # TODO SEMVER REGEX?
+    err "Please enter a valid version number next time."
   fi
+else
+  version=$1
 fi
 
 if [ $# -lt 2 ]; then
   read -r -p "Please enter the increment number [by default: 00]: " incr
   if [ -z "${incr}" ]; then
     incr="00"
   fi
+else
+  incr=$2
 fi
 
 current_branch=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')
 date=$(date "+%Y%m%d")
-dev_branch="release/v${version}-dev.${date}${incr}"
-canal_branch="release/v${version}-canal.${date}${incr}"
+dev_version="${version}-dev.${date}${incr}"
+canal_version="${version}-canal.${date}${incr}"
+dev_branch="release/v${dev_version}"
+canal_branch="release/v${canal_version}"
+
+if [ -n "$(git status --porcelain doc)" ]; then
+  echo "ERROR: Please commit your modifications to \"${current_branch}\" first"
```

Comment:
> the message can be something more explicit like: "There is unstaged changes in your worktree, Please commit your changes or stash them before creating the release."

I copied that exact message :D

> Also why does this only apply to the doc folder ?

Because I dumbly copied the gh-pages script and forgot to do the update.
This is now fixed.

> Wouldn't you use the function err() you just created ?

You're right. Fixed.